### PR TITLE
Expose eventemitter3 for ES6-style imports

### PR DESCRIPTION
**Pull request by delta62**
The `module.exports = someFunction` pattern is incompatible with ES6 imports. This change allows ES6-compatible environments (such as TypeScript or Babel) to import eventemitter3:

``` js
import { EventEmitter } from 'eventemitter3';
```

### CODE REVIEW

**Comment by 3rd-Eden** [comment]
Would it work if you did `EventEmitter.EventEmitter = EventEmitter`?  As that allows for better magnification.

**Comment by delta62** [comment]
Yes, I think that would be fine. I'll test it out this evening and confirm.

**Comment by lpinca** [comment]
Can't you use:
```js
import EventEmitter from 'eventemitter3';
```
This should work.

**Comment by fearphage** [comment]
> The module.exports = someFunction pattern is incompatible with ES6 imports.

This is not a true statement. This format is equivalent to the ES6 `exports default fn;`. It can be imported as @lpinca suggested. You can even combine the two. `import foo, {cat, pants} from 'bar';`.

**Comment by delta62** [comment]
The CJS `export =` and ES6 `export default` are not the same thing.

``` js
import EventEmitter from 'eventemitter3';`
```
is syntactic sugar for
``` js
import { default as EventEmitter } from 'eventemitter3'
```

As the module does not export a member named `default`, `EventEmitter === undefined` in the above code.

**Comment by delta62** [comment]
For reference, here is the [spec](http://www.ecma-international.org/ecma-262/6.0/#sec-imports-static-semantics-boundnames).

**Comment by lpinca** [comment]
@delta62 the last time I used babel
```js
import EventEmitter from 'eventemitter3';
```
worked fine. I don't know if something changed though.

**Comment by fearphage** [comment]
> CJS export = and ES6 export default are not the same thing.

Yeah, I didn't realize that at all. I stand corrected.

**Comment by delta62** [comment]
I have noticed that Babel seems to be okay with that syntax. I am using TypeScript as my transpiler, which doesn't like that.

The standard is somewhat hard to follow, but from what I understand TypeScript got it right this time.

**Comment by lpinca** [comment]
Gotcha. When you have time check if @3rd-Eden's suggestion work as it seems the cleanest approach.

**Comment by delta62** [comment]
Alright, I've tested @3rd-Eden's desired approach thoroughly and everything works using that syntax as well 👌

Also added a test case for good measure.

I was going to document this somewhere, but I'm not really sure where (or if) that should go in the README.

**Comment by lpinca** [comment]
Thank you.
